### ISSUE TITLE
dcadec test: undefined method `cached_download'

### ISSUE THREAD

**Issue by rhaglennydd** [Homebrew/homebrew-core#4513]
# Please follow the general troubleshooting steps first:
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
### Bug reports:

Running `brew test dcadec` causes Ruby to complain `undefined method `cached_download' for #<ResourceStageContext:0x007fa6bb830108>`. Here is the test code from dcadec.rb:

``` ruby
test do
  resource("sample").stage do |r|
    system "#{bin}/dcadec", r.cached_download # <-- The offending line
  end
end
```

The class `ResourceStageContext`, defined in resource.rb, doesn't define a method named `cached_download`, however the class `Resource`, defined in the same file, does.

**Comment by dunn** [comment]
Does changing `r.cached_download` to `resource("sample").cached_download` fix it?

**Comment by rhaglennydd** [comment]
Yes, that fixed it.

**Comment by dunn** [comment]
Great!  If you want to open a PR to fix the test, that'd be the fastest; otherwise we'll get to it when we can.  Take a look at https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md and let us know if you have questions!

**Comment by ilovezfs** [comment]
we probably want
```
    testpath.install resource("sample")
    system bin/"dcadec", "xll_71_24_96_768.dtshd"
```

**Comment by harshnarang8** [comment]
I think the issue was fixed through #10667, so it could be closed.
